lib/entities/environment-alias.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Environment alias instances
+ * @namespace EnvironmentAlias
+ * @description Environment aliases allow environments to be accessed through another identifier. You must have a V2 org and manually opt-in to this feature from the
+ *  environments settings page in the Contentful web app. If you do not have access to this feature, these methods will return 403 errors.
+ */
+import cloneDeep from 'lodash/cloneDeep'
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import enhanceWithMethods from '../enhance-with-methods'
+import {
+  createUpdateEntity
+} from '../instance-actions'
+
+/**
+ * @memberof EnvironmentAlias
+ * @typedef EnvironmentAlias
+ * @prop {Object} sys - System metadata
+ * @prop {string} sys.id - EnvironmentAlias id
+ * @prop {string} sys.type - Entity type
+ * @prop {Environment} environment - the environment this alias points to
+ * @prop {function(): Object} toPlainObject() - Returns this EnvironmentAlias as a plain JS object
+ */
+
+/**
+ * @memberof EnvironmentAlias
+ * @typedef EnvironmentAliasCollection
+ * @prop {number} total
+ * @prop {number} skip
+ * @prop {number} limit
+ * @prop {Array<EnvironmentAlias.EnvironmentAlias>} items
+ * @prop {function(): Object} toPlainObject() - Returns this EnvironmentAlias collection as a plain JS object
+ */
+function createEnvironmentAliasApi (http) {
+  return {
+    /**
+     * Sends an update to the server with any changes made to the object's properties. Currently, you can only change the id of the alias's underlying environment. See the example below.
+     * @memberof EnvironmentAlias
+     * @func update
+     * @return {Promise<EnvironmentAlias>} Object returned from the server with updated changes.
+     * @example
+     * const contentful = require('contentful-management')
+     *
+     * const client = contentful.createClient({
+     *   accessToken: '<content_management_api_key>'
+     * })
+     *
+     * client.getSpace('<space_id>')
+     * .then((space) => space.getEnvironmentAlias('<environment_alias_id>'))
+     * .then((alias) => {
+     *   alias.environment.sys.id = '<environment_id>'
+     *   return alias.update()
+     * })
+     * .then((alias) => console.log(`alias ${alias.sys.id} updated.`))
+     * .catch(console.error)
+     */
+    update: function () {
+      const environmentAlias = this
+      return createUpdateEntity({
+        http: http,
+        entityPath: 'environment_aliases',
+        wrapperMethod: wrapEnvironmentAlias,
+        headers: {'x-contentful-enable-alpha-feature': 'environment-aliasing'}
+      }).call(environmentAlias)
```

Comment:
any reason we are calling it here?
looking at other entities we don't do that https://github.com/contentful/contentful-management.js/blob/master/lib/entities/content-type.js#L60

I might be missing something though

Reply:
eek good catch. i copy/pasted from `locale`, which is formulated slightly differently and uses `.call(...)`... 🤦‍♂